platform/posix/transport/utest/openssl_utest.c

```diff
@@ -840,3 +865,28 @@ void test_Openssl_Recv_Network_Error( void )
     TEST_ASSERT_EQUAL( SSL_READ_WRITE_ERROR, bytesReceived );
     TEST_ASSERT_TRUE( bytesReceived <= 0 );
 }
+
+/**
+ * @brief Test that #Openssl_Recv returns zero bytes sent when #SSL_read returns
+ * an error to retry operation.
+ */
+void test_Openssl_Recv_Zero_Return_Value( void )
+{
+    int32_t bytesReceived;
+
+    /* Test when SSL_read returns SSL_ERROR_WANT_WRITE. */
+    opensslParams.pSsl = &ssl;
+    SSL_read_ExpectAnyArgsAndReturn( SSL_READ_WRITE_ERROR );
+
+    SSL_get_error_ExpectAnyArgsAndReturn( SSL_ERROR_WANT_WRITE );
+    bytesReceived = Openssl_Recv( &networkContext, opensslBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( 0, bytesReceived );
+
+    /* Test when SSL_read returns SSL_ERROR_WANT_READ. */
```

Comment:
The test below is already tested on line 852 and is duplicated here. It'd probably make sense to just move the above test into that block as well

Reply:
Thanks for catching. I have instead removed the older section as I prefer keeping the `0` and `-1` (or any negative value) return test cases as separate.